runtime/interpreter/value.go

```diff
@@ -6006,6 +6006,90 @@ func (AddressValue) SetMember(_ *Interpreter, _ LocationRange, _ string, _ Value
 	panic(errors.NewUnreachableError())
 }
 
+// AccountKeyValue
+
+type AccountKeyValue struct {
+	publicKey []byte
+}
+
+func NewAccountKeyValue(publicKey []byte) AccountKeyValue {
+	return AccountKeyValue{publicKey: publicKey}
+}
+
+func (AccountKeyValue) IsValue() {}
+
+func (AccountKeyValue) DynamicType(_ *Interpreter) DynamicType {
+	return AccountKeyDynamicType{}
+}
+
+func (v AccountKeyValue) Copy() Value {
+	return v
+}
+
+func (v AccountKeyValue) KeyString() string {
+	return v.String()
+}
+
+func (v AccountKeyValue) String() string {
+	return fmt.Sprintf("AccountKey")
+}
+
+func (AccountKeyValue) GetOwner() *common.Address {
+	// value is never owned
+	return nil
+}
+
+func (AccountKeyValue) SetOwner(_ *common.Address) {
+	// NO-OP: value cannot be owned
+}
+
+func (AccountKeyValue) IsModified() bool {
+	return false
+}
+
+func (AccountKeyValue) SetModified(_ bool) {
+	// NO-OP
+}
+
+func (v AccountKeyValue) Equal(other Value) BoolValue {
+	otherAccountKey, ok := other.(AccountKeyValue)
+	if !ok {
+		return false
+	}
+	return string(v.publicKey) == string(otherAccountKey.publicKey)
```

Comment:
This should also compare signature algorithm, hash algorithm, and weight, if this PR adds them